utils.py

```diff
@@ -35,14 +35,15 @@ def cache_response(response, uri=''):
     if 'ETag' not in response.headers:
         remove_response_from_cache(uri)
         return
-    cached_responses[uri] = response
+    database_helper.cache_response(uri, response)
 
 def caching_request_headers(uri):
-    return {'If-None-Match': cached_responses[uri].headers['ETag']} if uri in cached_responses else {}
+    result = database_helper.fetch_cached_response(uri)
+    return {'If-None-Match': result.headers['ETag']} if result is not None else {}
 
-def cached_response(uri):
-    return cached_responses[uri] if uri in cached_responses else None
+def fetch_cached_response(uri):
+    result = database_helper.fetch_cached_response(uri)
+    return result if result is not None else None
```

Comment:
no need for this check you can simply return database_helper.fetch_cached_response(uri)
If it is None then it will still be returned as None